### PR TITLE
new: add support listing `nodebalancer` `firewalls`

### PR DESCRIPTION
**Pull request by jriddle-linode**
## 📝 Description

**What does this PR do and why is this change necessary?**

Add support for `nodebalancers/{nb_id}/firewalls` endpoint

## ✔️ How to Test

**How do I run the relevant unit/integration tests?**

```bash
make ARGS='-run TestNodeBalancerFirewalls_List' fixtures
```

### CODE REVIEW

File: nodebalancer_firewalls.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	res, err := coupleAPIErrors(r.SetResult(NodeBalancerFirewallsPagedResponse{}).Get(e))
+	if err != nil {
+		return 0, 0, err
+	}
+	castedRes := res.Result().(*NodeBalancerFirewallsPagedResponse)
+	resp.Data = append(resp.Data, castedRes.Data...)
+	return castedRes.Pages, castedRes.Results, nil
+}
+
+// ListNodeaBalancerFirewalls returns a paginated list of Cloud Firewalls for nodebalancerID
```
Comment (lgarber-akamai) [inline nodebalancer_firewalls.go:31]: `ListNodeaBalancerFirewalls` -> `ListNodeBalancerFirewalls`

File: nodebalancer_firewalls.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+func (c *Client) ListNodeBalancerFirewalls(ctx context.Context, opts *ListOptions, nodebalancerID int) ([]Firewall, error) {
```
Comment (lgarber-akamai) [inline nodebalancer_firewalls.go:32]: To stay consistent with our other List functions, could `nodebalancerID` come before `opts`?

**Review by lgarber-akamai — CHANGES_REQUESTED** [review]
Just a few small changes, otherwise looks good!

**Review by lgarber-akamai — APPROVED** [review]
Great work! 

Fixtures pass as expected:

```bash
make ARGS="-run TestNodeBalancerFirewalls" fixtures
```

I also ran this manual test case and everything worked as expected:

```go
func main() {
	client := linodego.NewClient(nil)
	client.SetToken(os.Getenv("LINODE_TOKEN"))

	label := "fw-list-test"

	fw, err := client.CreateFirewall(
		context.Background(),
		linodego.FirewallCreateOptions{
			Label: label,
			Rules: linodego.FirewallRuleSet{
				InboundPolicy:  "DROP",
				OutboundPolicy: "DROP",
			},
		},
	)
	if err != nil {
		log.Fatal(err)
	}

	nb, err := client.CreateNodeBalancer(
		context.Background(),
		linodego.NodeBalancerCreateOptions{
			Label:      &label,
			Region:     "us-mia",
			FirewallID: fw.ID,
		},
	)
	if err != nil {
		log.Fatal(err)
	}

	nbFWs, err := client.ListNodeBalancerFirewalls(
		context.Background(),
		nb.ID,
		nil,
	)
	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(nbFWs[0].ID, fw.ID)
}
```

**Review by amisiorek-akamai — APPROVED** [review]
Tested locally (fixtures and Lena's manual guide), and works as expected!

**Pull request merged**